lib/ClangImporter/ClangImporter.cpp

```diff
@@ -7440,6 +7440,35 @@ static bool hasImportAsRefAttr(const clang::RecordDecl *decl) {
          });
 }
 
+// TODO: Move all these utility functions in a new file ClangImporterUtils.h
+// rdar://138803759
+static bool hasImmortalAtts(const clang::RecordDecl *decl) {
+  return decl->hasAttrs() && llvm::any_of(decl->getAttrs(), [](auto *attr) {
+           if (auto swiftAttr = dyn_cast<clang::SwiftAttrAttr>(attr))
+             return swiftAttr->getAttribute() == "retain:immortal" ||
+                    swiftAttr->getAttribute() == "release:immortal";
+           return false;
+         });
+}
+
+namespace swift {
+namespace importer {
```

Comment:
I think the convention of the codebase is to have these as part of the function name, e.g. `importer::isForeignReferenceTypeWithoutImmortalAttrs` in this case.